Engine/Src/Ancona/Framework/EntityFramework/AbstractSystem.hpp

```diff
@@ -42,6 +42,16 @@ class AbstractSystem
          * @brief Overridable destructor for systems.
          */
         virtual ~AbstractSystem() {}
+        
+        /**
+         * @brief Can be overridden to load in meta data files and configure settings
+         *        during the loading screen. 
+         *        
+         *        One example would be the collision system loading in collision types
+         *        from the project/collision-types.json file
+         */
+        virtual void LoadMetaData() { }
```

Comment:
This name is overly specific for the current use case. You should rename it to OnLoad.